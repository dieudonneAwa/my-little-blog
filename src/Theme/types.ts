export type Colors = {
  emerald: string;
  spacecadet: string;
  cultured: string;
  lightgrey: string;
  white: string;
  blueviolet: string;
  richblack: string;
};

export type Space = {
  none: string;
  nano: string;
  micro: string;
  mini: string;
  tiny: string;
  small: string;
  medium: string;
  large: string;
  big: string;
  huge: string;
  massive: string;
};

export type Font = {
  family: {
    poppins: string;
  };
  size: Space;
};

export type Radii = {
  none: string;
  small: string;
  medium: string;
  large: string;
  huge: string;
};

export type Article = {
  id: string;
  createdAt: string;
  title: string;
  content: string;
  createdBy: string;
};

export type Action = {
  type: string;
  payload?: any;
};

export type User = {
  id: string;
  name: string;
  avatar: string;
  createdAt: string;
};

export type Comment = {
  id: string;
  name: string;
  createdAt: string;
  content: string;
};

export type AppTheme = 'light' | 'dark';
