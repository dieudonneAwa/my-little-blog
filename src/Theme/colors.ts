const colors = {
  emerald: "#1EAE68",
  spacecadet: "#1b2546",
  cultured: "#F1F1F1",
  lightgrey: "#FBFBFB",
  white: "white",
  blueviolet: "#5621ab",
};

export default colors;
