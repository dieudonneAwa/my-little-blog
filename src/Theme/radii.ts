const radii = {
  none: '0',
  small: '3px',
  medium: '6px',
  large: '9px',
  huge: '50%',
}

export default radii
