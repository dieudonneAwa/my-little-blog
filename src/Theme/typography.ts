const fontfamilies = ["'Poppins', sans-serif"];

export const space = {
  none: "0rem",
  nano: "0.0625rem",
  micro: "0.125rem",
  mini: "0.25rem",
  tiny: "0.5rem",
  small: "0.75rem",
  medium: "1rem",
  large: "1.5rem",
  big: "2rem",
  huge: "3rem",
  massive: "4rem",
};

export const font = {
  family: {
    poppins: fontfamilies[0],
  },
  size: space,
};
