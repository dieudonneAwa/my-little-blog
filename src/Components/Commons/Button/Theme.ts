import { css } from "styled-components";
import { darken, lighten, transparentize } from "polished";
import { font, space } from "../../../Theme/typography";

// BASE
const buttonBase = css`
  font-weight: 500;
  transition: all 0.2s linear;
  min-width: fit-content;
  font-size: ${font.size.medium};

  &:hover {
    cursor: pointer;
    transform: scale(1);
    transition: all 0.5s cubic-bezier(0.075, 0.82, 0.165, 1);
  }
  &:not(:hover) {
    transform: scale(1);
    transition: all 0.6s cubic-bezier(0.075, 0.82, 0.165, 1);
  }
  @media (max-width: 767px) {
    padding: ${space.small};
  }
`;

// COLOR

const emerald = css`
  background: ${(p) => p.theme.colors.emerald};
  color: ${(p) => p.theme.colors.white};
  box-shadow: -1px -1px 8px ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.emerald))};
  border: 1px solid ${(p) => p.theme.colors.emerald};

  &:hover {
    background: ${(p) => lighten(0.05, p.theme.colors.emerald)};
    box-shadow: 0px 0px 12px
      ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.emerald))};
  }
`;

const spacecadet = css`
  border: 1px solid ${(p) => p.theme.colors.spacecadet};
  background: ${(p) => p.theme.colors.spacecadet};
  color: ${(p) => p.theme.colors.white};

  &:hover {
    background: ${(p) => darken(0.05, p.theme.colors.spacecadet)};
    box-shadow: 0px 0px 12px
      ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.spacecadet))};
  }
`;

const blueviolet = css`
  border: 1px solid ${(p) => p.theme.colors.blueviolet};
  background: ${(p) => p.theme.colors.blueviolet};
  color: ${(p) => p.theme.colors.white};

  &:hover {
    background: ${(p) => darken(0.05, p.theme.colors.blueviolet)};
    box-shadow: 0px 0px 12px
      ${(p) => transparentize(0.7, darken(0.2, p.theme.colors.blueviolet))};
  }
`;

const defaultcolor = css``;

const buttonColor: any = {
  emerald,
  spacecadet,
  default: defaultcolor,
  blueviolet,
};

// SIZE

const small = css`
  font-size: ${font.size.medium};
  padding: ${space.tiny} ${space.medium};
  border-radius: ${(p) => p.theme.radii.small};
`;

const medium = css`
  font-size: ${font.size.medium};
  padding: ${space.small} ${space.large};
  border-radius: ${(p) => p.theme.radii.small};
`;

const large = css`
  font-size: ${font.size.large};
  padding: ${space.medium} ${space.large};
  border-radius: ${(p) => p.theme.radii.medium};
`;

const buttonSize: any = {
  small,
  medium,
  large,
};

// SHAPE
const rounded = css``;

const square = css`
  border-radius: 0;
`;
const pill = css`
  border-radius: 50px;
`;

const buttonShape: any = {
  rounded,
  square,
  pill,
};

export { buttonBase, buttonColor, buttonSize, buttonShape };
