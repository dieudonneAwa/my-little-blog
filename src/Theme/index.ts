import { colors, darkThemeColors } from './colors';
import { Colors, Font, Radii, Space } from './types';
import { font, space } from './typography';
import radii from './radii';

export interface DefaultTheme {
  colors: Colors;
  font: Font;
  radii: Radii;
  space: Space;
}

const theme: DefaultTheme = {
  colors,
  font,
  radii,
  space,
};

const darkTheme: DefaultTheme = {
  ...theme,
  colors: darkThemeColors,
};

export { theme, darkTheme };
