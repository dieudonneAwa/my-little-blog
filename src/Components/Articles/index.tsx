import { AxiosResponse } from 'axios';
import { useEffect } from 'react';
import { Action, Article } from '../../Theme/types';
import { Box } from '../Commons/Box';
import Spinner from '../Commons/Spinner';
import ArticleCard from './components/ArticleCard';

export type Props = {
  fetchArticles: () => Promise<AxiosResponse<any>>;
  articles: Article[];
  loading: boolean;
  setArticle: (article: Article) => Action;
};

const ArticlesPage = ({ fetchArticles, articles, setArticle, loading }: Props) => {
  useEffect(() => {
    (async () => {
      await fetchArticles();
    })();
  }, []);

  return (
    <>
      {articles?.map((article) => (
        <ArticleCard setArticle={setArticle} key={article.id} article={article} />
      ))}
      {loading && (
        <Box display="flex" justifyContent="center" alignItems="center" height="500px">
          <Spinner />
        </Box>
      )}
    </>
  );
};

export default ArticlesPage;
