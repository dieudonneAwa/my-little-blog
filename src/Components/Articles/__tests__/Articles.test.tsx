/* eslint-disable import/no-extraneous-dependencies */
import { mount } from 'enzyme';
import Articles, { Props } from '../index';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../Theme';

const props: Props = {
  fetchArticles: jest.fn().mockResolvedValue([
    {
      id: '1',
      title: 'Test title',
      createdAt: '',
      content: 'Test content',
      createdBy: 'Awa',
    },
  ]),
  articles: [
    {
      id: '1',
      title: 'Test title',
      createdAt: '',
      content: 'Test content',
      createdBy: 'Awa',
    },
  ],
  loading: false,
  setArticle: jest.fn().mockReturnValue({
    type: 'SET_ARTICLE',
    payload: {
      id: '1',
      title: 'Test title',
      createdAt: '',
      content: 'Test content',
      createdBy: 'Awa',
    },
  }),
};

describe('Articles page', (): void => {
  let wrapper: any;

  beforeEach((): void => {
    wrapper = mount(
      <ThemeProvider theme={theme}>
        <Articles {...props} />
      </ThemeProvider>
    );
  });

  it('should render Articles page component', (): void => {
    expect(wrapper).toMatchSnapshot();
  });
});
