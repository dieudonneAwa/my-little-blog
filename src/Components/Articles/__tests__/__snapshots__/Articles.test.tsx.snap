// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles page should render Articles page component 1`] = `
<Component
  theme={
    Object {
      "colors": Object {
        "blueviolet": "#5621ab",
        "cultured": "#F1F1F1",
        "emerald": "#1EAE68",
        "lightgrey": "#FBFBFB",
        "richblack": "#091A28",
        "spacecadet": "#1b2546",
        "white": "white",
      },
      "font": Object {
        "family": Object {
          "poppins": "'Poppins', sans-serif",
        },
        "size": Object {
          "big": "2rem",
          "huge": "3rem",
          "large": "1.5rem",
          "massive": "4rem",
          "medium": "1.1rem",
          "micro": "0.15rem",
          "mini": "0.3rem",
          "nano": "0.065rem",
          "none": "0rem",
          "small": "0.9rem",
          "tiny": "0.6rem",
        },
      },
      "radii": Object {
        "huge": "50%",
        "large": "9px",
        "medium": "6px",
        "none": "0",
        "small": "3px",
      },
      "space": Object {
        "big": "2rem",
        "huge": "3rem",
        "large": "1.5rem",
        "massive": "4rem",
        "medium": "1.1rem",
        "micro": "0.15rem",
        "mini": "0.3rem",
        "nano": "0.065rem",
        "none": "0rem",
        "small": "0.9rem",
        "tiny": "0.6rem",
      },
    }
  }
>
  <ArticlesPage
    articles={
      Array [
        Object {
          "content": "Test content",
          "createdAt": "",
          "createdBy": "Awa",
          "id": "1",
          "title": "Test title",
        },
      ]
    }
    fetchArticles={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    loading={false}
    setArticle={[MockFunction]}
  >
    <styled.div
      px="1rem"
    >
      <div
        className="sc-bdvvaa hroOLu"
      >
        <ArticleCard
          article={
            Object {
              "content": "Test content",
              "createdAt": "",
              "createdBy": "Awa",
              "id": "1",
              "title": "Test title",
            }
          }
          key="1"
          setArticle={[MockFunction]}
        >
          <Styled(styled.div)
            as="article"
            bg="white"
            borderColor="cultured"
            borderStyle="solid"
            borderWidth="1px"
            idnumber={0.3333333333333333}
            m="0.5rem auto"
            onClick={[Function]}
            p={
              Array [
                "1.5rem 1.5rem",
                "2rem 3rem",
              ]
            }
            width={
              Array [
                "100%",
                "100%",
                "700px",
              ]
            }
          >
            <article
              className="sc-bdvvaa sc-hKwCoD fdhzRd cjyoeq"
              onClick={[Function]}
              width={
                Array [
                  "100%",
                  "100%",
                  "700px",
                ]
              }
            >
              <styled.h2>
                <h2
                  className="sc-eCImvq fAnUyj"
                >
                  Test title
                </h2>
              </styled.h2>
              <span>
                Test content
              </span>
              <br />
              <styled.small>
                <small
                  className="sc-jRQAMF eaOvtV"
                >
                  Posted Invalid Date by 484639
                </small>
              </styled.small>
            </article>
          </Styled(styled.div)>
        </ArticleCard>
      </div>
    </styled.div>
  </ArticlesPage>
</Component>
`;
